scipy/interpolate/_bsplines.py

```diff
@@ -744,6 +744,11 @@ def make_interp_spline(x, y, k=3, t=None, bc_type=None, axis=0,
         except TypeError:
             raise ValueError("Unknown boundary condition: %s" % bc_type)
 
+    if not -y.ndim <= axis < y.ndim:
```

Comment:
Is it okay if I just asarray(y) in front of here, or should I copy this block to each special and the main cases?

Reply:
Yes, asarray here would be fine, I think.